MapboxNavigation/InstructionsBannerViewLayout.swift

```diff
@@ -0,0 +1,125 @@
+import UIKit
+
+extension InstructionsBannerView {
+    
+    func commonInit() {
+        setupViews()
+        setupLayout()
+        setupAvailableBounds()
+    }
+    
+    func setupViews() {
+        backgroundColor = .clear
+        
+        let maneuverView = ManeuverView()
+        maneuverView.backgroundColor = .clear
+        maneuverView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(maneuverView)
+        self.maneuverView = maneuverView
+        
+        let distanceLabel = DistanceLabel()
+        distanceLabel.translatesAutoresizingMaskIntoConstraints = false
+        distanceLabel.adjustsFontSizeToFitWidth = true
+        distanceLabel.minimumScaleFactor = 20.0 / 22.0
+        addSubview(distanceLabel)
+        self.distanceLabel = distanceLabel
+        
+        let primaryLabel = PrimaryLabel()
+        primaryLabel.translatesAutoresizingMaskIntoConstraints = false
+        primaryLabel.allowsDefaultTighteningForTruncation = true
+        primaryLabel.adjustsFontSizeToFitWidth = true
+        primaryLabel.numberOfLines = 1
+        primaryLabel.minimumScaleFactor = 26.0 / 30.0
+        primaryLabel.lineBreakMode = .byTruncatingTail
+        addSubview(primaryLabel)
+        self.primaryLabel = primaryLabel
+        
+        let secondaryLabel = SecondaryLabel()
+        secondaryLabel.translatesAutoresizingMaskIntoConstraints = false
+        secondaryLabel.adjustsFontSizeToFitWidth = true
+        secondaryLabel.allowsDefaultTighteningForTruncation = true
+        secondaryLabel.numberOfLines = 2
+        secondaryLabel.minimumScaleFactor = 20.0 / 26.0
+        secondaryLabel.lineBreakMode = .byTruncatingTail
+        addSubview(secondaryLabel)
+        self.secondaryLabel = secondaryLabel
+        
+        let dividerView = UIView()
+        dividerView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(dividerView)
+        self.dividerView = dividerView
+        
+        let separatorView = UIView()
+        separatorView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(separatorView)
+        self.separatorView = separatorView
+    }
+    
+    func setupLayout() {
+        // Distance label
+        distanceLabel.centerXAnchor.constraint(equalTo: maneuverView.centerXAnchor, constant: 0).isActive = true
```

Comment:
@frederoni can you explain briefly why you chose to setup constraints here for this view here instead of in the storyboard?

Reply:
It's part of #645, componentization of the UI. Also noted in the description of the PR
> Extract instructions banner from storyboard into an isolated component